aiida/common/datastructures.py

```diff
@@ -104,6 +104,14 @@ class CodeInfo(DefaultFieldsAttributeDict):
 
         code.x cmdline_params[0] cmdline_params[1] ... < stdin > stdout
 
+    * ``folder_path``: (optional) the path, in string, of the folder where to
+      run the code. If it is specified, the code will be called as:
+
+        (cd folder_path; code.x ...)
+
+      The outputs will be produced in this folder.
```

Comment:
Maybe it should therefore also be made clear that the various retrieve lists such as `retrieve_list` should be updated accordingly to specify paths that include this relative path.

Reply:
Absolutely good point!